terraform/modules/base/main.tf

```diff
@@ -0,0 +1,115 @@
+data "azurerm_client_config" "main" {}
+
+data "azurerm_resource_group" "main" {
+    name = "Cohort28_CamBod_ProjectExercise"
+}
+
+resource "azurerm_cosmosdb_mongo_database" "db" {
+	name = var.mongo_database_name
+	resource_group_name = data.azurerm_resource_group.main.name
+	account_name = var.db_account_name
+}
+
+resource "azurerm_service_plan" "main" {
+    name                = "${var.prefix}-cb-todo-app-sp-tf"
+    location            = data.azurerm_resource_group.main.location
+    resource_group_name = data.azurerm_resource_group.main.name
+    os_type             = "Linux"
+    sku_name            = "B1"
+}
+
+resource "azurerm_linux_web_app" "main" {
+    name                = "${var.prefix}-${var.app_service_name}"
+    location            = data.azurerm_resource_group.main.location
+    resource_group_name = data.azurerm_resource_group.main.name
+    service_plan_id     = azurerm_service_plan.main.id
+
+    site_config {
+        application_stack {
+          docker_image_name = "cameronb9/todo-app:prod"
+          docker_registry_url = "https://index.docker.io"
+        }
+    }
+
+    app_settings = {
+		"DOCKER_REGISTRY_SERVER_URL"          = "https://cameronb9"
+		"FLASK_APP"                           = "todo_app/app"
+		"FLASK_ENV"                           = "production"
+		"LOGIN_DISABLED"                      = var.login_disabled
+		"GITHUB_OAUTH_URL"                    = "https://github.com/login/oauth"
+		"HOMEPAGE_URL"                        = "https://${var.prefix}-${var.app_service_name}.azurewebsites.net"
```

Comment:
Probably not worrying about here, but if you do need to compose the same arguments into the same format several times in one resource, we could look at using a [Local](https://developer.hashicorp.com/terraform/language/values/locals)
(e.g. so that this doesn't break if someone updates the resource name on line 22 but forgets to update this)